pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -297,16 +297,21 @@ contract StablePhantomPool is
 
         _upscaleArray(balances, scalingFactors);
         (uint256 virtualSupply, uint256[] memory balancesWithoutBpt) = _payProtocolFeesBeforeJoinExit(balances);
+
         uint256 upscaledAmountGiven = _upscale(swapRequest.amount, scalingFactors[isGivenIn ? indexIn : indexOut]);
 
         (uint256 amp, ) = _getAmplificationParameter();
+        uint256 preJoinExitInvariant = StableMath._calculateInvariant(amp, balancesWithoutBpt);
+        // bptAmount is the amount swapped in or out
+        uint256 bptAmount = upscaledAmountGiven;
```

Comment:
This is a bit magical tbh, I'd rather we always set bptAmount to the correct value instead of setting it to a potentially wrong value and then fixing it if that's the case

Reply:
Changing the arguments so that this goes away